src/PingController.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\PingController;
+
+use ErrorException;
+use Exception;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Psr\Log\NullLogger;
+use RuntimeException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Throwable;
+use function call_user_func;
+use function in_array;
+use function restore_error_handler;
+use function set_error_handler;
+use const E_DEPRECATED;
+use const E_USER_DEPRECATED;
+
+final class PingController
+{
+    private $test;
+    private $logger;
+
+    public function __construct(callable $test = null, LoggerInterface $logger = null)
+    {
+        $this->test = $test;
+        $this->logger = $logger;
+    }
+
+    public function __invoke(Request $request) : Response
+    {
+        if ($this->test) {
+            $this->ensureLogger();
+
+            set_error_handler(
```

Comment:
this is probably something for a bootstrap phase, more than for a single controller. Being process-wide, it could only be set for an application rather than in a library, although a library can provide an implementation

Reply:
The application should have its own error handler already, this switches it for the duration of the check to allow _all_ responses to come from this controller.

Reply:
Makes sense to cover all errors coming from inside this PHP process, although there will always be cases where the error comes from elsewhere like nginx, the bootstrap, etc; the happy path is the important one as any error that makes it different from `200 OK` `pong` works for healthchecks